src/Components/MainTableView/TableView.tsx

```diff
@@ -182,12 +253,26 @@ class TableView extends React.Component<{token: string, selectedSchemaName: stri
         }
       }
 
-      this.setState({tableContentData: result.tuples, tableRecordTotal: result.total_count, errorMessage: '', isLoading: false})
+      this.setState({tableContentData: result.tuples, totalNumOfTuples: result.total_count, errorMessage: '', maxPageNumber:  Math.ceil(result.total_count / this.state.numberOfTuplesPerPage), isLoading: false})
     })
     .catch(error => {
       this.setState({tableContentData: [], errorMessage: 'Problem fetching table content: ' + error, isLoading: false})
     })
   }
+
+  setPageNumber(pageNumber: number) {
+    console.log(pageNumber)
+    console.log(this.state.maxPageNumber);
+    if (pageNumber < 1 || pageNumber > this.state.maxPageNumber) {
+      throw Error('Invalid pageNumber ' + pageNumber + ' requested');
+    }
```

Comment:
Could we just make this an `assert` and catch assertion errors if needed?

Reply:
Yep sure, to be honest this shoudl have been assert error cause this is more of a bad coding error then actual runtime error. My mistake

Reply:
Typsecript doesn't really have assert, at least not without making another function and definining aassert ourselves

Reply:
The assert that typescript has documented is misleading as in the assert function is not actually defined, and the user have to defined it themselves. Here is the playground code that the assert docs linked to. https://www.typescriptlang.org/play?ssl=14&ssc=56&pln=14&pc=1#code/PTAEHEEsDcFMDtQCkCG0UGUDGAnSAHAFwHIBnUAMwBtYAPSAI0iskIE8AaUV0LFRBrFApQAcwD24gCbcpsEYXHCpUgLAAoEKBwBXeIUgBbIVgAWsLAGtyi0G3E6cvaUNvoWUlISH3Hw0qQ6hkSQ4vCkAHQaGloAKuakQig4rmz4kHxUVGy8KFmwMigBsDgGYeQAFOJOkPDoePyEpACUwvBqmmCE5jnJQqSGeVSUelhl4aAA7qYZptookImgJTjV5PI42VPm8DFgvk71kCgMNORSYcSEoIOEZqA6+KC20152Dsu0+BaEUep7oAAci5nMZyJNWHMRBRRuNKNVQBdaqJnjNyA5ruIKKihAxxLQOADWGRcvlCsVSm0ZMTyGgFlQTjQppCMW1uMFqr9op1QAB1WDEFJicTI55KOQUWpCBxOcSTRDdByiUwRUDxRaIiwMlK0gEw+BjUKIaazfyJUo2UxvbpCOj4HWkI2kmgdLTuHRCDWEXSwABcGjkWG1Qn1hrCZpKhAq7r9D3glngcvgzV9EYtoBjAG5uVpgZMpgKhToljw3HlIJ5vDjnmlpdj+Mt4EEA1q+s5wtdBmxBBhvciAPI4QFBQQ4VOkPvwFEAH1ATcMo+z6iK5qj7G+WJuKG7sF7eCng+HC5KoAAvOfQAAiCf70SX5pLgG8yFq2vYPBEUAAZgiAHYuDazhyJQVBym0eRsI65B8IgxauAksAAuusDotiobjDYSiSqIjjSjo1yvNcgFYCCiyqhg4iEjy0wIO8OigKY4hwLKzHVkcjK4rAoH5gAtHRuSIIk8FeES1xWuQgi0fAySrJMBSUKshjCKAN6igi86jmKAIAFbFtcIiqVOfwaF2PaTqIh4jiUj48gAmh8MEPEsK6Rk66FGphW6WEIOHJI0sAoaAWIAgcNbfJaqw6MqrKhbUFArPJJFyFwFAInQKDBDQAKxG+uAENcCZypa1poiMBpwhCWTaLAhCOIgKAAhpJ4Qoq+HVvAAUyLYKAqGFgW2IB+B+cY3g4AC0DHNWJxMUkFLXIG2peEaxnqO54aGDoVAGPg2QVLQqb8JwdgHfAbCtAA3hooBpmutabrQZ4XpeTU4PeS7XS5pQVMhm45Oep5Xi9b3ctdKS1TgiAPQAVHYS4AL45mAACCc1ubCHnCEKjqnMink5d8EA6MkUikAC6WZX6yE8b5OAk58WCwJ+Nq9BQ8VjNWJH6KswzUHKAI0YgPCcwYTb9aYkXKtWa3wCtWgAGJpbQGU7bAXByQJTmzauaPlRjtjSTgskAg2CBBIiSZBaxBjGP6-zqAtbacxOQUdf2FBy5AOATh7cBHqOpCpgAjKAs4AEwh9+EcACwRwArDZDtCtL3CkP2KgVENOAZameiFfKKY3eQmcZSnoDB7OX5x0uichuj4aLAAQlxcoK44GfDTn8aJvnB1zUXw2l+XoDhxX0c2XEpWkIxm0yFgbUXPK1am0pihBxHlezrH3KLGnUhVK77ue97MCwH7JQtEuYSwG7Hte4QPun1ZXvj2A6rgswww2ogc9EeYjZBCSfA4gAiMCZBOLw-VxBrwrtvUgTduKtxwPva+h874PzPl7B8Ggr43yPvfE+GDSAvzVKVDUIh7RGBPOGUg4hjBBWxIBeKXhcKoRujrMM4QiSIHxruPKn4fy-lAHxXwGtJTtCCm1Qw1RcQ5BSD1UUNoAQpBoEUIQiZvABz2ACUwhBCD4ADiAOQ0BTjiFEJEQwGRVg0IoL8EihhgDIVIHwwgwB+CJlGMiHijjnE8S-DxX8wAgA